extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/DevConsoleProcessor.java

```diff
@@ -940,7 +940,7 @@ public Object apply(EvalContext ctx) {
                 for (Path sourcePaths : sourcesDir) {
                     List<String> packages = sourcePackagesForRoot(sourcePaths);
                     if (!packages.isEmpty()) {
-                        sourcePackagesByDir.put(sourcePaths.toAbsolutePath().toString(), packages);
+                        sourcePackagesByDir.put(sourcePaths.toAbsolutePath().toString().replace("\\", "/"), packages);
```

Comment:
So somehow when we consume the keys from the Map, we have `/` in the paths even on Windows? Do you have handy the place where this is consumed?

Reply:
The map is consumed in the `main.html` ([here](https://github.com/quarkusio/quarkus/blob/main/extensions/vertx-http/deployment/src/main/resources/dev-templates/main.html#L147-L152)) and subsequently used by the JS function `openInIde` later in that HTML.

Depending on the IDE type used the call either tries set a `window.location` (for `client` type IDEs) or it invokes the server (for `server` type IDEs) using AJAX where the call ends up in [OpenIdeHandler.java](https://github.com/quarkusio/quarkus/blob/main/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/OpenIdeHandler.java).

The path is actually not relevant for the `server` type IDEs since only the fully-qualified class name is sent to the server which the resolves a proper path for the OS anyway.

As for the `client` type IDEs, it seems that only _vscode_ is currently supported and there the path pattern `vscode://file/{0}:{1}` is used and it uses internal also forward slash to specify paths on windows (i.e. `vscode-file://vscode-app/c:/Users/ExampleUser/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html`).

The open IDE flow can be tested when running the DEVUI locally on the ArC Beans page (http://localhost:8080/q/dev/io.quarkus.quarkus-arc/beans) by clicking the link underneath a user-defined bean.